scene/gui/popup_menu.cpp

```diff
@@ -62,6 +62,11 @@ Size2 PopupMenu::_get_contents_minimum_size() const {
 		const_cast<PopupMenu *>(this)->_shape_item(i);
 
 		Size2 icon_size = items[i].get_icon_size();
+		if (items[i].icon_max_width > 0 && icon_size.width > items[i].icon_max_width) {
+			icon_size.height = icon_size.height * items[i].icon_max_width / icon_size.width;
+			icon_size.width = items[i].icon_max_width;
+		}
```

Comment:
The same should be applied to global menu icons here (it's using child `PopupMenu`s to build global menu):
https://github.com/godotengine/godot/blob/c29866dbc05694594b9be68b8d451816c7a0889d/scene/gui/menu_bar.cpp#L231

Something like:
```c++
Ref<Texture2D> icon = p_child->get_item_icon(i);
if (icon.is_valid()) {
	int icon_max_width = p_child->get_item_icon_max_width(i);
	Size2i icon_size = icon->get_size();
	if (icon_max_width > 0 && icon_size.width > icon_max_width) {
		icon_size.height = icon_size.height * icon_max_width / icon_size.width;
		icon_size.width = icon_max_width;
		icon = icon->duplicate();
		icon->resize(icon_size);
	}
}
DisplayServer::get_singleton()->global_menu_set_item_icon(p_menu_name, index, icon);
```
or maybe `item_icon_max_width` can be added to the global menu API.

Reply:
Either way works, though I'm not a fan of duplicating the resource. But is it actually possible to overflow the size of a native menu? I'd think they have a hard limit and would scale the icon already.

Reply:
We also don't have a resize method on `Texture2D`, we'd need to create an `ImageTexture` from the image, or something like that. So I'm still wondering if it's actually possible to overflow native menus. I don't have a Mac to test it myself.

Reply:
It is possible to use any size of the image in the macOS menu, and it will stretch the menu, but there's already hard-coded `resize(16, 16, Image::INTERPOLATE_LANCZOS);` in the macOS DisplayServer. So it should be good as is (probably `item_icon_max_width` should replace this hardcoded size in the future, just in case someone wants bigger menu icons).

<img width="588" alt="Screenshot 2023-03-30 at 14 24 32" src="https://user-images.githubusercontent.com/7645683/228821361-ed2e4891-a744-4b8d-9db4-976f8bea1489.png">